website/source/docs/secrets/pki/index.html.md

```diff
@@ -0,0 +1,686 @@
+        <span class="param-flags">optional</span>
+        If set, clients can request any CN. Useful in some
+        circumstances, but make sure you understand whether it
+        is appropriate for your installation before enabling it.
+        Defaults to `false`.
+      </li>
+      <li>
+        <span class="param">allow_ip_sans</span>
+        <span class="param-flags">optional</span>
+        If set, clients can request IP Subject Alternative
+        Names. Unlike CNs, no authorization checking is
+        performed except to verify that the given values
+        are valid IP addresses. Defaults to `true`.
+      <li>
+        <span class="param">server_flag</span>
+        <span class="param-flags">optional</span>
+        If set, certificates are flagged for server use.
+        Defaults to `true`.
+      </li>
+      <li>
+        <span class="param">client_flag</span>
+        <span class="param-flags">optional</span>
+        If set, certificates are flagged for client use.
+        Defaults to `true`.
+      </li>
+      <li>
+        <span class="param">code_signing_flag</span>
+        <span class="param-flags">optional</span>
+        If set, certificates are flagged for code signing
+        use. Defaults to `false`.
+      </li>
+      <li>
+        <span class="param">key_type</span>
+        <span class="param-flags">optional</span>
+        The type of key to generate for generated private
+        keys. Currently, `rsa` and `ec` are supported.
+        Defaults to `rsa`.
+      </li>
+      <li>
+        <span class="param">key_bits</span>
+        <span class="param-flags">optional</span>
+        The number of bits to use for the generated keys.
+        Defaults to `2048`; this will need to be changed for
+        `ec` keys. See https://golang.org/pkg/crypto/elliptic/#Curve
+        for an overview of allowed bit lengths for `ec`.
+      </li>
+    </ul>
+  </dd>
+
+  <dt>Returns</dt>
+  <dd>
+    A `204` response code.
+  </dd>
+</dl>
+
+#### GET
+
+<dl class="api">
+  <dt>Description</dt>
+  <dd>
+    Queries the role definition.
+  </dd>
+
+  <dt>Method</dt>
+  <dd>GET</dd>
+
+  <dt>URL</dt>
+  <dd>`/pki/roles/<name>`</dd>
+
+  <dt>Parameters</dt>
+  <dd>
+     None
+  </dd>
+
+  <dt>Returns</dt>
+  <dd>
+
+    ```javascript
+    {
+        "data": {
+            "allow_any_name": false,
+            "allow_ip_sans": true,
+            "allow_localhost": true,
+            "allow_subdomains": false,
+            "allow_token_displayname": false,
+            "allowed_base_domain": "example.com",
+            "client_flag": true,
+            "code_signing_flag": false,
+            "key_bits": 2048,
+            "key_type": "rsa",
+            "lease": "6h",
+            "lease_max": "12h",
+            "server_flag": true
+        }
+    }
+    ```
+
+  </dd>
+</dl>
+
+
+#### DELETE
+
+<dl class="api">
+  <dt>Description</dt>
+  <dd>
+    Deletes the role definition.
```

Comment:
We should add something like: "Deleting a role does not revoke the certificates previously issued under this role."